main/lsp/LSPIndexer.cc

```diff
@@ -103,7 +103,7 @@ bool LSPIndexer::canTakeFastPath(const std::vector<std::shared_ptr<core::File>>
             return false;
         } else {
             const auto &oldFile = getOldFile(fref, *initialGS, evictedFilesRef);
-            if (oldFile.sourceType == core::File::Type::Package) {
+            if (oldFile.sourceType == core::File::Type::Package && oldFile.source() != f->source()) {
```

Comment:
This is doing full text string equality on a package file? We don't do this anywhere else. What's the justification here?

Reply:
For non-package files, we use a content hash to determine if the file has changed in such a way that requires the slow path.

This hash is not sufficient for package files, which need to trigger a slow path whenever imports/exports are changed (which, to the hashing process, is just another send).

Rather than implement a wholly unique hashing system for packages, I did the dumb thing: See if the source text has changed at all.

The main reason I needed this is so that opening a `__packager.rb` file in the editor or retypechecking it to determine code actions does not automatically trigger a slow path.

Reply:
I can add a comment above this line.